functions/ocr/app/ocr_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ocr
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"cloud.google.com/go/pubsub"
+	"cloud.google.com/go/storage"
+	"cloud.google.com/go/translate"
+	vision "cloud.google.com/go/vision/apiv1"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"golang.org/x/text/language"
+)
+
+const (
+	menuName = "menu.jpg"
+	signName = "sign.png"
+)
+
+var (
+	bucketName      string
+	imageBucketName string
+)
+
+// TestMain sets up the config rather than using the config file
+// which contains placeholder values.
+func TestMain(m *testing.M) {
+	ctx := context.Background()
+	tc, ok := testutil.ContextMain(m)
+	if !ok {
+		log.Fatalf("testutil.ContextMain failed")
```

Comment:
Should not use `log.Fatal` in tests - it stops the whole test suite. If `ContextMain` failed, it's probably because `GOLANG_SAMPLES_PROJECT_ID` is not set, in which case we shouldn't fail.

```
log.Print("GOLANG_SAMPLES_PROJECT_ID is unset. Skipping.")
return
```

Reply:
Done